packages/react/fast-components-react-base/src/data-grid/data-grid-cell.tsx

```diff
@@ -0,0 +1,176 @@
+import { DataGridCellClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import {
+    DataGridCellHandledProps,
+    DataGridCellProps,
+    DataGridCellUnhandledProps,
+} from "./data-grid-cell.props";
+import { DataGridContext, DataGridContextType } from "./data-grid-context";
+
+class DataGridCell extends Foundation<
+    DataGridCellHandledProps,
+    DataGridCellUnhandledProps,
+    {}
+> {
+    public static defaultProps: Partial<DataGridCellProps> = {
+        rowData: null,
+        columnDefinition: null,
+        columnIndex: 0,
+        managedClasses: {},
+    };
+
+    public static displayName: string = "DataGridCell";
+    public static contextType: React.Context<DataGridContextType> = DataGridContext;
+
+    protected handledProps: HandledProps<DataGridCellHandledProps> = {
+        managedClasses: void 0,
+        rowData: void 0,
+        columnDefinition: void 0,
+        columnIndex: void 0,
+    };
+
+    private focusTarget: React.RefObject<any> = React.createRef();
+    private rootElement: React.RefObject<any> = React.createRef();
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactNode {
+        if (
+            isNil(this.context) ||
+            isNil(this.context.dataGridProps) ||
+            isNil(this.props.columnDefinition)
+        ) {
+            return null;
+        }
+
+        const unhandledProps: React.HTMLAttributes<HTMLElement> = {
+            role: "gridcell",
+            tabIndex: this.isFocusCell() ? 0 : -1,
+            onKeyDown: this.handleKeyDown,
+            onFocus: this.handleFocus,
+        };
+
+        if (!isNil(this.props.columnDefinition.cell)) {
+            return this.props.columnDefinition.cell(
+                this.props,
+                this.generateClassNames(),
+                this.props.columnDefinition.columnDataKey,
+                this.rootElement,
+                this.focusTarget,
+                unhandledProps
+            );
+        } else {
+            return this.defaultCellRenderFunction(
+                this.props,
+                this.generateClassNames(),
+                this.props.columnDefinition.columnDataKey,
+                this.rootElement,
+                this.focusTarget,
+                unhandledProps
+            );
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidMount(): void {
+        if (this.isDesiredFocusCell() && !isNil(this.rootElement.current)) {
+            this.rootElement.current.focus();
+        }
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        return super.generateClassNames(
+            classNames(this.props.managedClasses.dataGridCell)
+        );
+    }
+
+    /**
+     * The default function that renders an unstyled content display
+     */
+    public defaultCellRenderFunction = (
```

Comment:
could this be `unstyledCellRenderer`? also, is this being public intended? how should it be used if so?

Reply:
Well it does pass the class names so "unstyled" doesn't seem right (I took out "unstyled from the comment").  There won't be a "styled" version of this is the MSFT branch.  How about "renderCell"?  

As for the public bit - we've had cases where we have had similar customizable render functions (ie. select) which partners have used because they wanted to decorate ui elements for things such as generating telemetry and ended up duplicating the entire default render function to do that.  By having the default renderer public they can create a custom renderer that can optionally call the default one to create a cell node that they can modify.  That way they don't have to copy the base function and pick up changes without needing to recopy.  I'm also changing it to a public static function.